doc/dev/static_type_checking_cheat_sheet.md

```diff
@@ -430,3 +430,64 @@ def sanitize(log) -> Sanitized:
 
 def print_log(log: Sanitized) -> None: ...
 ```
+
+### **kwargs
+
+- When working with Python and handling keyword arguments, it's essential to ensure type safety while retaining flexibility. One way to achieve this is by using the `Any` type along with the `**kwargs` parameter. This allows you to accept and process any keyword arguments while preserving type hinting and validation.
+
+```python
+from typing import Any, Dict
+
+def process_data(**kwargs: Dict[str, Any]) -> None:
+    # Your code here
+    pass
+
+# Or
+
+def process_data(**kwargs: Any) -> None:
+    # Your code here
+    pass
+```
+
+### Getter and Setter
+
+- Property return type should match with setter value type
+
+```python
+# for example:
+
+@property
+def func(self) -> Optional[SparkJobEntry]:
+    pass
+
+@func.setter
+def func(self, value: Union[Dict[str, str], SparkJobEntry, None]):
+    pass
+
+# After being fixed, it should be:
+
+@property
+def func(self) -> Optional[Union[Dict[str, str], SparkJobEntry]]:
+    pass
+
+@func.setter
+def func(self, value: Optional[Union[Dict[str, str], SparkJobEntry]]):
+    pass
+```
+
+
+### Returning Any from function
+
+- In some cases, mypy will complain about returning Any from a function with a return declaration of bool, for example
```

Comment:
It's likely mypy is complaining about returning Any in this scenario because it ignores any "private" functions, i.e. _to_rest_object().

Also, surprised that mypy would complain here unless the warn-return-any option is turned on: https://mypy.readthedocs.io/en/stable/error_code_list2.html#check-that-function-does-not-return-any-value-no-any-return